src/metabase/driver/mongo/query_processor.clj

```diff
@@ -211,16 +234,45 @@
 
 ;;; ### filter
 
+;; omf: 'max' date in a date range is next day at 00:00:00.000
+(defn- resolve-between [basetype minvalue maxvalue]
+  (cond (= basetype :type/DateTime)
+    {$gte minvalue
+     $lt (t/plus maxvalue (t/days 1))}
+  :else
+    {$gte minvalue
+     $lte maxvalue})
+)
+
+;; omf: a single day ends before next day at 00:00:00.000
+(defn- resolve-equal [basetype value unit]
+  (cond (= basetype :type/DateTime)
```

Comment:
should just use `if` here, not `cond`. Also `isa?` to check the type of something